Create build connectors document

johannahemminger opened the pull request.

Commit:
Create build connectors document

Adam-Dupaski commented:
Please start assigning your PRs to yourself for the component that you own.

johannahemminger commented:
@Adam-Dupaski ah, I assigned you because I thought this was part of our proofreading workshop.

johannahemminger commented:
@Adam-Dupaski should have added you as a Reviewer, I see!

Adam-Dupaski commented:
Looks like a new doc? At any rate, you own these PRs. :)

Adam-Dupaski commented:
I don't need to be a reviewer either, just tag me in the PR.

Commit:
Update tags formatting

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+1. Create a [Mendix project](refguide/new-project/) for your connector.
+2. Create a Module to contain your connector. To do this, right-click anywhere in the **App Explorer** outside of existing modules and choose **Add Module**.
+![](/attachments/connectors/add-module.png)
```
Can you crop this screenshot so it's not as big? I understand the editor should be in there, but the image can be smaller, and you can also consider adding a mouse pointer to the screenshot.

Also, you will need to review syntax on how to insert an attachment in a Markdown list: indentation, spacing, and an extra space between the number-period and the first letter.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+---
+title: "Understand and Build Mendix Connectors"
+parent: "Creating-Content"
+#menu_order: Enter the position of the document in the category or under the parent; number by 10 (for first), 20, 30, etc. for easy ordering of other documents in the future if necessary; don't add brackets or quotation marks; if no number is added, the system will add an extremely high number to order the documents, which means that if you only want a document to appear at the top, you only hav ...
```
If not using certain metadata, you can remove it

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+## 1 Introduction
+
+Welcome to the world of Mendix connectors! Mendix connectors allow you to connect your Mendix application to other systems. You can find them on the Mendix Marketplace, or add them to the Marketplace for others to use. For more information on content you can create for the Marketplace, see [Creating Content](/appstore/creating-content).
```
Cross-reference for `[Mendix Marketplace]` would be useful - general Marketplace docs or connectors category?

Adam-Dupaski replied:
If we put this in the **Creating Content** category, that cross-reference should not be necessary. 

This is one of the main things to decide on for these docs - is there stakeholder agreement over this location?

Adam-Dupaski replied:
We can discuss location in a meeting about publishing this.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+---
+title: "Understand and Build Mendix Connectors"
```
`Understand` should be assumed and is thus not necessary in the title - `Build Mendix Connectors` is much better, stronger, to the point

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+* What connectors are and how they work
+* How to build a Mendix connector
+
+**We aim to reach the following target audiences:**
```
We need to decide if this audience is specific enough to even include. My impression is these audiences are all broad and assumed, and unless there's a specific use case for a persona/level, this does not need to be added.

johannahemminger replied:
Agree, I removed it

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+Before starting this how-to, make sure you have completed the following prerequisites:
+
+* Read the [introductory blog on the Mendix Connector Kit](https://www.mendix.com/blog/introducing-mendix-connector-kit/)
+* Read [Share Marketplace Content](https://docs.mendix.com/appstore/general/share-app-store-content)
```
'How to Share...'

'doc about Mendix integration' - that's not just one doc, it's a category of ~20 docs; recommending/requiring users read all those is not specific enough. Can you cite a specific how-to? Or can you recommend to `have a look` or `check out` ?

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+* Read the [introductory blog on the Mendix Connector Kit](https://www.mendix.com/blog/introducing-mendix-connector-kit/)
+* Read [Share Marketplace Content](https://docs.mendix.com/appstore/general/share-app-store-content)
+* Read the [doc about Mendix integration](howto/integration/) and check out the [evaluation guide post on Mendix integration](https://www.mendix.com/evaluation-guide/app-capabilities/integration/)
+* Read how to [build microflow actions with Java](https://docs.mendix.com/howto/extensibility/howto-connector-kit) 
```
The cross-reference text for this type of CR should be the exact title of the doc - please fix capitalization. 

You also need a relative link, not an absolute link to the doc for all cross-references.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+* Read [Share Marketplace Content](https://docs.mendix.com/appstore/general/share-app-store-content)
+* Read the [doc about Mendix integration](howto/integration/) and check out the [evaluation guide post on Mendix integration](https://www.mendix.com/evaluation-guide/app-capabilities/integration/)
+* Read how to [build microflow actions with Java](https://docs.mendix.com/howto/extensibility/howto-connector-kit) 
+* View a [YouTube tutorial on building a Mendix connector](https://www.youtube.com/watch?v=wTQJ1MiXAow)
```
Cross-reference text should be the specific title of the video in this case (does not matter it's on Youtube):

`View the [Build a Connector in Mendix](https://www.youtube.com/watch?v=wTQJ1MiXAow) tutorial`

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+## 2 What Are Connectors?
+
+Mendix connectors can be used in your Mendix project to simplify connecting to existing third-party systems or other Mendix applications. 
```
We no longer use "Mendix project," we use "Mendix app."

There are 3 instances of "Mendix in this sentence - probably 2 too many.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+Mendix connectors can be used in your Mendix project to simplify connecting to existing third-party systems or other Mendix applications. 
+
+These connectors are implemented as Mendix modules with a focus on connecting to other systems. The connectors can make use of any of the platform supported protocols like SOAP, REST, OData, or [Data Hub](/data-hub/). They can also be used to package a Java library provided by the other system to make the connection. These Java library functions can be exposed via Java actions as [custom microflo ...
```
"platform-supported"

Can cross-references be added for "SOAP, REST, OData"?

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+These connectors are implemented as Mendix modules with a focus on connecting to other systems. The connectors can make use of any of the platform supported protocols like SOAP, REST, OData, or [Data Hub](/data-hub/). They can also be used to package a Java library provided by the other system to make the connection. These Java library functions can be exposed via Java actions as [custom microflo ...
+
+The charts below provide overviews of use cases and categoriation for connectors:
```
Typo

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+The charts below provide overviews of use cases and categoriation for connectors:
+
+| Mendix to Mendix| Solution | 
```
What does "Mendix to Mendix" mean? Is that clear for users?

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+| Mendix to Mendix| Solution | 
+| ----------- | ----------- | 
+| Platform-supported protocols | Connect to two or more Mendix applications using platform supported facilities: SOAP webservices, REST webservices, ODATA or Data Hub. <br>**References**</br>* [Publish Web Services](https://docs.mendix.com/refguide/published-web-services)<br>* [Consumed Web Services](https://docs.mendix.com/refguide/consumed-web-services)<br>* [Published REST Services](https://do ...
```
These "References" links should be incorporated into the description sentence as cross-references. Otherwise, we have duplicated info, not efficient to read, and we do not use lists of "references" like this.

The links need to be relative.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+| Mendix to Mendix| Solution | 
+| ----------- | ----------- | 
+| Platform-supported protocols | Connect to two or more Mendix applications using platform supported facilities: SOAP webservices, REST webservices, ODATA or Data Hub. <br>**References**</br>* [Publish Web Services](https://docs.mendix.com/refguide/published-web-services)<br>* [Consumed Web Services](https://docs.mendix.com/refguide/consumed-web-services)<br>* [Published REST Services](https://do ...
+| Unsupported protocols   | Build a module to connect either via alternative protocols or by encapsulating one of the platform supported protocols. <br>**References**</br> * [Java actions](https://docs.mendix.com/refguide/java-actions)<br>* [JavaScript actions](https://docs.mendix.com/refguide/javascript-actions)   |  |
```
Same thing here for need to use cross-references instead of another/detached list.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+| Mendix to Third-Party | Solution 
+| ----------------- | ----------------------- |
+| Platform-supported protocols      | Connect a Mendix application to one or more third party applications using platform supported facilities: SOAP webservices, REST webservices or ODATA.       | 
```
"web service" is 2 words

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+| ----------------- | ----------------------- |
+| Platform-supported protocols      | Connect a Mendix application to one or more third party applications using platform supported facilities: SOAP webservices, REST webservices or ODATA.       | 
+| Partner-supported protocols (eQ)   | Use eQ supported facilities: Webservices, REST, OData | 
+| Unsupported protocols   | Build a module to connect either via alternative protocols, or by encapsulating one of the platform supported protocols. | 
```
"eQ-supported", "platform-supported"

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+### 2.1 Getting Mendix Connectors
+
+Many existing Mendix connectors can be downloaded from the [Mendix Marketplace](https://marketplace.mendix.com/) directly into your project. Depending on your use case, your company's Mendix Marketplace could also have a variety of Mendix connector modules available. See an [explanation of available Mendix connectors](/appstore/connectors/).
```
Can you give an example of a connector, or a deep-link that takes users to the connector category?

"your company's Mendix Marketplace" = private content: can you find a cross-reference in Marketplace docs for this?

Please follow standard syntax for the "explanation" cross-reference text and doc title.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+Many existing Mendix connectors can be downloaded from the [Mendix Marketplace](https://marketplace.mendix.com/) directly into your project. Depending on your use case, your company's Mendix Marketplace could also have a variety of Mendix connector modules available. See an [explanation of available Mendix connectors](/appstore/connectors/).
+
+Because connectors expose data via custom Microflow activities, you can find them in the Studio Pro **Toolbox** to implement when building your application logic. Then, you can drag them where you want to use the connector functionality.
```
"microflow"

"Toolbox" - cross-reference?

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+### 2.2 Deploying Connectors
+
+Once you deploy an application, the connector will automatically deploy with it. You can also set a few runtime or node specific configurations (via the application interface or constants).
```
"You can also set a few runtime or node specific configurations (via the application interface or constants)."

"runtime- or node-specific"

This isn't very clear - can you add some explanation or specify a use case?

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+#### 2.2.1 Future Versions of Connectors
+
+Another approach to connecting to external systems is by creating connector applications that wrap those systems with OData APIs. This approach will be discussed in a future version of this document.
```
"future version of this document" or a different document?

johannahemminger replied:
Good question, I'm not sure yet if we'll add a section to this document or put it somewhere else. I will try to be more clear

johannahemminger replied:
"This approach will be discussed in the future, either here or in a linked document."

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+#### 2.2.1 Future Versions of Connectors
+
+Another approach to connecting to external systems is by creating connector applications that wrap those systems with OData APIs. This approach will be discussed in a future version of this document.
+
```
"future version of this document" or a different document?

What part of https://docs.mendix.com/appstore/general/share-app-store-content can you cross-reference specifically for "securing Marketplace content" in this context?

Also, best practice for cross-reference links (also to ensure easier conversion to new system): `/appstore/general/share-app-store-content` (and not `appstore/general/share-app-store-content/`)

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+When creating a new Mendix connector, it is advised to have a project in which you both develop and test your connector. This means that you will have a module that contains all your connector logic, along with one or more additional modules that assist in the testing and development. The logic in the additional modules are not required for the operation of the module and therefore need to be sep ...
+
+### 3.1 Setting Up Your Project
```
To clarify, "project" should be changed to "app" throughout the doc.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+### 3.1 Setting Up Your Project
+
+1. Create a [Mendix project](refguide/new-project/) for your connector.
+2. Create a Module to contain your connector. To do this, right-click anywhere in the **App Explorer** outside of existing modules and choose **Add Module**.
```
"module"

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+* Microflows
+* Pages
+* Java actions
+* Theming (from the `themesource` folder) *[only for Mx 9+]*
```
Please review standard for citing version numbers in the SG

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+* Specified resource files (from the `resources` folder)
+* Specified related Java files (from the modules sub folder in the `javasource` folder)
+
+#### 3.3.1 Exporting as an *.mpk* File
```
I wouldn't recommend extra formatting in headings, so italics won't be necessary - it'd have to be verified upon build, and extra formatting in headings is generally unnecessary.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+#### 3.3.1 Exporting as an *.mpk* File
+
+1. Ensure that version number in the version folder corresponds with Marketplace version number.
```
Unclear what "Marketplace version number" means - the version number of the component in the Marketplace being exported?

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+#### 3.3.1 Exporting as an *.mpk* File
+
+1. Ensure that version number in the version folder corresponds with Marketplace version number.
+2. Generate the export module package using your [Gradle script](https://github.com/ako/CsvServices/blob/cd219e71249c194bca26b374716b88628237a6dd/build.gradle#L72)
```
Missing period.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+
+1. Ensure that version number in the version folder corresponds with Marketplace version number.
+2. Generate the export module package using your [Gradle script](https://github.com/ako/CsvServices/blob/cd219e71249c194bca26b374716b88628237a6dd/build.gradle#L72)
+3. Build/export release candidate with versioning and put it in the Marketplace `DIST` folder.
```
Where can this "DIST" folder be found, and should it be bolded instead of getting code formatting?

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,132 @@
+1. Ensure that version number in the version folder corresponds with Marketplace version number.
+2. Generate the export module package using your [Gradle script](https://github.com/ako/CsvServices/blob/cd219e71249c194bca26b374716b88628237a6dd/build.gradle#L72)
+3. Build/export release candidate with versioning and put it in the Marketplace `DIST` folder.
+4. Ensure your version is aligned with the Marketplace version number. Versioning in Marketplace starts at 1.0.0. See the *Adding New Marketplace Content* section in [Share Marketplace Content](appstore/general/share-app-store-content)/. You cannot use Marketplace to distribute your connector if it is below version 1.0.0.
```
Does this duplicate step 1?

For the cross-reference, review the standard syntax and add an anchor - I would write, in parentheses: `(for more information, see the *Adding New Marketplace Content* section in [How to Share Marketplace Content](/appstore/general/share-app-store-content#adding)).`

Commit:
Update most changes (not chart or screenshot yet)

Looking into the chart and screenshot changes and technical verification on suggested sections.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,127 @@
+
+### 3.4 Distributing the Connector
+
+To share your connector with other developers and the wider community you can publish it in the Mendix Marketplace. While publishing you may choose to publish it for everyone or just for your company. In case you choose the second option it will only be available to other developers belonging to the same company. For more information on how to publish to the Marketplace, see [Share Marketplace Co ...
```
Could use some commas in these sentences.

Adam-Dupaski replied:
Can you find a section with an anchor for the private Marketplace cross-reference?

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,127 @@
+
+Once you publish the connector and someone imports it in their Mendix app, all of these elements will be placed in the same location from which they are exported.
+
+Distribute the module to whomever wants to use it (share the file or upload to Marketplace).
```
"by following these steps:"

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,127 @@
+Once you publish the connector and someone imports it in their Mendix app, all of these elements will be placed in the same location from which they are exported.
+
+Distribute the module to whomever wants to use it (share the file or upload to Marketplace).
+1. Keep the release notes ready.
```
These steps are covered by the How to Share doc in the MG, so I don't know that a duplicated section here is necessary.

johannahemminger replied:
I see what you mean, but I think since it's small I'll leave it so users don't have to jump around. I'm also linking that doc here.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,127 @@
+3. Create the release in Marketplace using the release from GitHub.
+
+**Architectural impact:**
+Because connectors are exported as modules and then published on the private/public Mendix Marketplace, they are as decoupled as possible from your development application. This is needed because a copy of the module will be imported in other applications when they download the module from the Marketplace. The implementing application might need to configure some constants/add specific microflows ...
```
"private/public" can probably just be deleted.

Who is "they" in the second sentence?

`constants/add specific microflows/use pages or snippets/configure specifics` - this needs to be broken up into clauses with commas.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,127 @@
+
+## 4 Read More
+
+This document introduced connectors and gave you a step-by-step outline on how to build one. An upcoming document will contain more detailed information and good practices for this process.
```
Without any proper links to existing docs, this section is unnecessary. If you want to announce forthcoming docs, I'd put that statement in the intro.

Adam-Dupaski commented:
@johannahemminger My review is done. Please address all comments, resolve them, and push changes to your branch. Once you have done that, we can have a meeting to finalize publication.

johannahemminger commented:
Thanks for the review @Adam-Dupaski . I am gathering the remaining info for the Good/Best/Recommended Practices document now. I think it might make sense to publish them together, so I will try to get it into a PR soon.

Adam-Dupaski commented:
@johannahemminger I think it'd be best to do 1 doc at a time and prioritize feedback on this doc. There are a lot of comments to sort through but the changes should be relatively quick. Once we decide on where to publish this doc, we can publish it and then you can get back to the child docs. Do you agree?

johannahemminger commented:
@Adam-Dupaski sure sounds good. Thanks! I'll let you know once comments are sorted through and branch updated.

Commit:
Make most of the rest of the changes

Commit:
Final changes and updates

Commit:
Final final change

johannahemminger commented:
@Adam-Dupaski all changes have been made and all comments resolved, let's chat sometime!

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+Welcome to the world of Mendix connectors! Mendix connectors allow you to connect your Mendix application to other systems. You can find them on the [Mendix Marketplace](https://marketplace.mendix.com/), or add them to the Marketplace for others to use. For more information on content you can create for the Marketplace, see [Creating Content](/appstore/creating-content).
+
+**This how-to will teach you the following:**
+* What connectors are and how they work
```
Line break between intro sentence and bulleted list is best practice.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+* What connectors are and how they work
+* How to build a Mendix connector
+
+Check back for a link to a future document about recommended practices for building connectors.
```
Use

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+Before starting this how-to, make sure you have completed the following prerequisites:
+
+* Read the blog post [Introducting the Mendix Connector Kit](https://www.mendix.com/blog/introducing-mendix-connector-kit/)
+* Read how to [Share Marketplace Content](/appstore/general/share-app-store-content)
```
Move "How to" into cross-reference text (title of doc) - see latest SG comment in our Slack.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+
+Connectors can be used in your app to simplify connecting to existing third-party systems or other Mendix applications. 
+
+These connectors are implemented as Mendix modules with a focus on connecting to other systems. The connectors can make use of any of the platform-supported protocols like [SOAP](https://en.wikipedia.org/wiki/SOAP), [REST](https://en.wikipedia.org/wiki/Representational_state_transfer), [OData](https://www.odata.org/), or [Data Hub](/data-hub/). They can also be used to package a Java library prov ...
```
By cross-references for these protocols I meant our docs, not third-party sites. Third-party site links will become a maintainability issue if they break.

johannahemminger replied:
How can I reference these so they're not too long? Like SOAP, I would link to /refguide/consumed-app-services#soap, but then wouldn't I have to say "see the SOAP section in the Consumed App Services document"?

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+
+The charts below provide overviews of use cases and categorization for connectors:
+
+### Connecting Mendix Apps to Mendix Apps
```
These sub-section headers need numbering: 2.1, etc.

Add a line break between header and first line (table).

johannahemminger replied:
I have tried something else, I hope it renders OK.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+
+### Connecting Mendix Apps to Mendix Apps
+| | Solution | 
+| ---------------- | ----------- | 
```
Can you add something for the first column name to match "Solution"? Otherwise looks incomplete.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+
+### 2.1 Getting Mendix Connectors
+
+Many existing Mendix [Connectors](/appstore/connectors) can be downloaded from the [Mendix Marketplace](https://marketplace.mendix.com/) directly into your app. Depending on your use case, your company's private Mendix Marketplace could also have a variety of Mendix connector modules available. For more information on the distinction between public and private Marketplace content, see the Getting ...
```
Please review best practice for cross-referencing a section of a doc rather than the entire doc. The section name should be the cross-reference text, and the doc name should be italicized afterward.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+
+Another approach to connecting to external systems is by creating connector applications that wrap those systems with OData APIs. This approach will be discussed in the future, either here or in a linked document.
+
+In addition, it is important to protect the IP of the connector module, particularly when reselling those modules. Applying IP protection to connector modules will be discussed in the future, either here or in a linked document. For more information on securing Marketplace content, see [Sharing Marketplace Content](/appstore/general/share-app-store-content).
```
Please use correct title/verb of doc instead of "Sharing".

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+
+### 3.1 Setting Up Your App
+
+1.  Create a [Mendix app](/refguide/new-project) for your connector.</br>
```
You do not need breaks after every line in a numbered list.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+
+1.  Create a [Mendix app](/refguide/new-project) for your connector.</br>
+2.  Create a module to contain your connector. To do this, right-click anywhere in the **App Explorer** outside of existing modules and choose **Add Module**.</br>
+    ![](attachments/connectors/add-module.png)</br>
```
Still need work on this attachment insertion in a list - search other docs for how this is done. You do not need the break. You just need line breaks on either side of the doc and appropriate indentation.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+* Microflows
+* Pages
+* Java actions
+* Theming (from the `themesource` folder) *[only for Mendix Studio Pro 9 and above]*
```
Do you need code formatting around the folder names in this section or bold (for UI elements)?

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+* Specified related Java files (from the modules sub folder in the `javasource` folder)
+
+#### 3.3.1 Exporting as an .mpk File
+
```
Best practice is to have an intro sentence before a numbered or bulleted list, not just start a new section right away with a list. Several sections in your doc need this attention.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+
+#### 3.3.1 Exporting as an .mpk File
+
+1. Ensure your version is aligned with the version number that appears on the Marketplace version. Versioning in Marketplace starts at 1.0.0 (for more information, see the *Adding New Marketplace Content* section in [How to Share Marketplace Content](/appstore/general/share-app-store-content#adding)). You cannot use Marketplace to distribute your connector if it is below version 1.0.0.
```
Review best practice for cross-referencing a section of a doc rather than the entire doc.

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,128 @@
+    * Create new release
+3. Create the release in Marketplace using the release from GitHub.
+
+See [Share Marketplace Content](/appstore/general/share-app-store-content) for more information on adding content to the Marketplace.
```
There are several cross-references to this entire doc. In general, I would only cross-reference the entire doc once, and then specific sections as necessary. You can then add a Read More section at the bottom of this doc with a final link to the entire doc.

Commit:
Update doc with further changes (except table)

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,137 @@
+---
+title: "Build Mendix Connectors"
```
For this doc to be published under the https://docs.mendix.com/appstore/creating-content/sol-solutions-guide sub-category, it needs updated metadata:

- Title should be consistent, so use a gerund: "Building Connectors" ("Mendix" not necessary)
- category: "Creating Content"
- menu_order: 3

Reference https://raw.githubusercontent.com/mendix/docs/development/content/appstore/creating-content/sol-solutions-guide.md for a model.

Commit:
Changes and xref updates

Commit:
Fix xref

Adam-Dupaski commented on content/appstore/creating-content/connector-guide-build.md:
```diff
@@ -0,0 +1,153 @@
+---
+title: "Building Connectors"
+linktitle: "Building Connectors"
+parent: "Creating Content"
```
`category`

Commit:
Change formatting
